docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,417 @@
+
+**GraphQL Query**\
+After getting column names array (`columnNames`) for the entity we create the graphQL query payload 
+```
+query = $"{{{entityName.ToLowerInvariant()}(first: {First}) {{items {{ {string.Join(" ", columnNames)} }}}}}}"
+````
+We execute a **POST** query against the GraphQL base URL to calculate time elapsed for health check and use this value to compare with threshold according to the [Overall Health Calculation](#overall-health-calculation) and get the status as `Healthy/Unhealthy`.
+
+CURL Command
+```
+curl --request POST \
+  --url <baseUrl>/<runtime.graphql.path> \
+  --header 'Content-Type: application/json' \
+  --header 'User-Agent: insomnia/10.3.0' \
+  --data '{
+	"query": "{UserTable(first: 4) {items { content id }}}"
+}'
+```
+
+## Output Sample
+After executing the above queries for DB and entities, we create the below sample for comprehensive Health Check Report.
+```
+{
+  "status": "Unhealthy/Healthy",
+  "version": "1.2.10",
+  "app-name": "dab_oss_1.2.10",
+  "configuration": {
+    "http": true,
+    "https": true,
+    "rest": true,
+    "graphql": true,
+    "telemetry": true,
+    "caching": true,
+    "mode": "development"
+  },
+  "checks": [
+    {
+      "name": "database-name",
+      "status": "Healthy",
+      "tags": ["data-source", "mssql"],
+      "data": {
+          "duration-ms": 10,
+          "threshold-ms": 100
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Healthy",
+      "tags": ["endpoint", "rest"],
+      "data": {
+          "duration-ms": 2,
+          "threshold-ms": 10
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Healthy",
+      "tags": ["endpoint", "graphql"]
+      "data": {
+          "duration-ms": 20,
+          "threshold-ms": 50
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Unhealthy",
+      "tags": ["endpoint", "rest"]
+      "exception": "{exception-message-here}",
+      "data": {
+          "duration-ms": 20,
+          "threshold-ms": 20
+      }
+    },
+    {
+      "name": "<entity-name>",
+      "status": "Unhealthy",
+      "tags": ["endpoint", "graphql"]
+      "exception": "{exception-message-here}",
+      "data": {
+          "duration-ms": 20,
+          "threshold-ms": 10
+      }
+    },
+  ]
+}
+```
+
+## Test Scenarios
+
+Health Check result scenarios for different cases of Health checks for DataSource, Rest and GraphQL entities to be enabled or disabled.
+
+![Class Diagram](dataSourceHealthCheckTestScenarioFlowchart.png)
```

Comment:
Thank you for adding these diagrams! Helps get an overall idea of the different conditions! 💯